pkg/morph/client/container/wrapper/container.go

```diff
@@ -17,51 +18,85 @@ var (
 	errUnsupported = errors.New("unsupported structure version")
 )
 
-// Put saves passed container structure in NeoFS system
-// through Container contract call.
+// Put marshals container, and passes it to Wrapper's Put method
+// along with sig.Key() and sig.Sign().
 //
-// Returns calculated container identifier and any error
-// encountered that caused the saving to interrupt.
-func (w *Wrapper) Put(cnr *container.Container, pubKey, signature []byte) (*container.ID, error) {
-	if cnr == nil || len(pubKey) == 0 || len(signature) == 0 {
+// Returns error if container is nil.
+func Put(w *Wrapper, cnr *container.Container, sig *pkg.Signature) (*container.ID, error) {
+	if cnr == nil {
 		return nil, errNilArgument
 	}
 
-	args := client.PutArgs{}
-	args.SetPublicKey(pubKey)
-	args.SetSignature(signature)
-
-	id := container.NewID()
-
 	data, err := cnr.Marshal()
 	if err != nil {
 		return nil, fmt.Errorf("can't marshal container: %w", err)
 	}
 
+	err = w.Put(data, sig.Key(), sig.Sign())
+	if err != nil {
+		return nil, err
+	}
+
+	id := container.NewID()
 	id.SetSHA256(sha256.Sum256(data))
-	args.SetContainer(data)
 
-	return id, w.client.Put(args)
+	return id, nil
 }
 
-// Get reads the container from NeoFS system by identifier
-// through Container contract call.
+// Put saves binary container with its key and signature
+// in NeoFS system through Container contract call.
 //
-// If an empty slice is returned for the requested identifier,
-// storage.ErrNotFound error is returned.
-func (w *Wrapper) Get(cid *container.ID) (*container.Container, error) {
-	if cid == nil {
-		return nil, errNilArgument
+// Returns calculated container identifier and any error
+// encountered that caused the saving to interrupt.
+//
+// If TryNotary is provided, call notary contract.
+func (w *Wrapper) Put(cnr, key, sig []byte) error {
```

Comment:
PutBinary

Reply:
There are no other `Put...`

Reply:
No, here we use same naming, the difference only in argument types.